src/main/webapp/authorization.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * The ID of the login button
+ * @type {String}
+ */
+const LOGIN_BTN = 'login-btn';
+
+// GCP Generated Github 'OAuth' configuration
+let config = {
+  apiKey: 'AIzaSyAR88Giah8cCEAvT_zDSIREWvgIIAeS8yY',
+  authDomain: 'step2020-279820.firebaseapp.com',
+};
+firebase.initializeApp(config);
+
+/**
+ * Function called when clicking the Login/Logout button.
+ */
+function toggleSignIn() {
+  // if not logged in
+  if (!firebase.auth().currentUser) {
+    let provider = new firebase.auth.GithubAuthProvider();
+    // sign in
+    firebase.auth().signInWithPopup(provider).then(function(result) {
```

Comment:
nit: I feel that this code is self-explanatory
```suggestion
    firebase.auth().signInWithPopup(provider).then(function(result) {
```